app/lib/shared/utils.dart

```diff
@@ -47,10 +47,19 @@ Future<T> withTempDirectory<T>(Future<T> func(Directory dir),
   });
 }
 
+/// Returns null if version is invalid or cannot be normalized.
 String canonicalizeVersion(String version) {
+  if (version == null || version.trim().isEmpty) {
+    return null;
+  }
   // NOTE: This is a hack because [semver.Version.parse] does not remove
   // leading zeros for integer fields.
-  final v = semver.Version.parse(version);
+  semver.Version v;
+  try {
+    v = semver.Version.parse(version);
+  } catch (_) {
```

Comment:
Could we settle with catching `FormatException`, isn't that what is thrown?

It's probably fine to get an error of we're passing null or something else... Or if semver was null...